test/cppsuite/tests/search_near_02.cxx

```diff
@@ -0,0 +1,376 @@
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "test_harness/test.h"
+#include "test_harness/util/api_const.h"
+#include "test_harness/workload/random_generator.h"
+
+using namespace test_harness;
+
+/*
+ * In this test, we want to verify search_near with prefix enabled returns the correct key.
+ * During the test duration:
+ *  - N threads will keep inserting new random keys
+ *  - M threads will execute search_near calls with prefix enabled using random prefixes as well.
+ * Each search_near call with prefix enabled is verified using the default search_near.
+ */
+class search_near_02 : public test_harness::test {
+    public:
+    search_near_02(const test_harness::test_args &args) : test(args) {}
+
+    void
+    populate(test_harness::database &database, test_harness::timestamp_manager *,
+      test_harness::configuration *config, test_harness::workload_tracking *) override final
+    {
+        /*
+         * The populate phase only creates empty collections. The number of collections is defined
+         * in the configuration.
+         */
+        int64_t collection_count = config->get_int(COLLECTION_COUNT);
+
+        logger::log_msg(
+          LOG_INFO, "Populate: " + std::to_string(collection_count) + " creating collections.");
+
+        for (int64_t i = 0; i < collection_count; ++i)
+            database.add_collection();
+
+        logger::log_msg(LOG_INFO, "Populate: finished.");
+    }
+
+    void
+    insert_operation(test_harness::thread_context *tc) override final
+    {
+        /* Each insert operation will insert new keys in the collections. */
+        logger::log_msg(
+          LOG_INFO, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
+
+        /* Helper struct which stores a pointer to a collection and a cursor associated with it. */
+        struct collection_cursor {
+            collection_cursor(collection &coll, scoped_cursor &&cursor)
+                : coll(coll), cursor(std::move(cursor))
+            {
+            }
+            collection &coll;
+            scoped_cursor cursor;
+        };
+
+        /* Collection cursor vector. */
+        std::vector<collection_cursor> ccv;
+        uint64_t collection_count = tc->db.get_collection_count();
+        uint64_t collections_per_thread = collection_count / tc->thread_count;
+
+        /* Must have unique collections for each thread. */
+        testutil_assert(collection_count % tc->thread_count == 0);
+        for (int i = tc->id * collections_per_thread;
```

Comment:
this is a little confusing to read. Maybe setting variable `offset = tc->id * collections_per_thread;` might read better.

Reply:
Done in 366d3f77a82939ffbcc2b3e6fd510be3d5858416